### DISCUSSION TITLE
redirection with Nginx

### DISCUSSION THREAD

**Discussion by NoipName** [thorsten/phpMyFAQ#2747] [Q&A] [answered]
Hello all,
Today i try install PhpMyFAQ with Nginx and i have some weird problem.
When i'm in admin interface and try to go another section "FAQ" "PARAMETER" "SAVE" ... I have kick to the main page direcly :V 

I need to do something for the rewrite with Nginx ?
The redirection need some specific right to work ?
All my file phpmyfaq have 777 right (to test) and the owner is root:root

On Debian 11
PHP 8.2
PhpMyFAQ 3.2.3
Admin local account (not Azure)

Scenario : Main page -> Admin interface -> Parameter -> Have kickout and return to the main page

There is some log:
192.168.10.1 - - [07/Dec/2023:13:01:52 +0100] "GET /admin/index.php HTTP/2.0" 200 20823 "https://phpmyfaq.test.com/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 Edg/119.0.0.0"
192.168.10.1 - - [07/Dec/2023:13:01:52 +0100] "GET /assets/dist/backend.js?1701950512 HTTP/2.0" 200 4961306 "https://phpmyfaq.test.com/admin/index.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 Edg/119.0.0.0"
192.168.10.1 - - [07/Dec/2023:13:01:52 +0100] "GET /admin/session.keepalive.php?lang=fr HTTP/2.0" 200 1726 "https://phpmyfaq.test.com/admin/index.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 Edg/119.0.0.0"
192.168.10.1 - - [07/Dec/2023:13:01:53 +0100] "GET /admin/index.php?action=ajax&ajax=dashboard&ajaxaction=version HTTP/2.0" 200 66 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 Edg/119.0.0.0"
192.168.10.1 - - [07/Dec/2023:13:01:53 +0100] "GET /admin/index.php?action=ajax&ajax=dashboard&ajaxaction=user-visits-last-30-days HTTP/2.0" 200 1025 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 Edg/119.0.0.0"
192.168.10.1 - - [07/Dec/2023:13:01:56 +0100] "GET /admin/?action=config HTTP/2.0" 200 11590 "https://phpmyfaq.test.com/admin/index.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 Edg/119.0.0.0"

Thanks you guys :D

**Comment by thorsten** [comment]
Looks like your PHP session management does not work correctly.

**Reply by NoipName** [reply]
I have these parameter in the vhost
![image](https://github.com/thorsten/phpMyFAQ/assets/145665195/e47d810a-894f-40f3-8d47-e2517a43f085)
can this make incident ?

**Reply by thorsten** [reply]
No, the path to the session files in the php.ini files has to be correct, maybe you can take a look here: https://copyprogramming.com/howto/php-session-files-permissions

**Reply by NoipName** [reply]
After analyse, i have all access with php account management is all okay, i think the problem from the rewrite of .htaccess with nginx

have you some exemple for the rewrite compalibitiy with Nginx ? I tried asked ChatGPT for the rewrite but that not work 

There is a rewrite:

location / {
        try_files $uri /index.php;
        alias /srv/www/phpmyfaq/;
        index login_page.php;
    }

    location ~ \.php$ {
        include snippets/fastcgi-php.conf;
        fastcgi_pass unix:/var/run/php/php8.2-fpm.sock;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        include fastcgi_params;
    }

    location ~ /\.ht {
        deny all;
    }

    add_header X-Frame-Options SAMEORIGIN;

    location ~* \.(ico|pdf|flv|jpg|jpeg|png|gif|js|css|swf)$ {
        add_header Cache-Control "max-age=84600, no-store";
        add_header Pragma "no-cache";
    }

    error_page 404 /index.php?action=404;


    location ~ /show-categories\.html$ {
        rewrite ^(.*)$ /index.php?action=show last;
    }

    location ~ /search\.html$ {
        rewrite ^(.*)$ /index.php?action=search last;
    }

    location ~ /addcontent\.html$ {
        rewrite ^(.*)$ /index.php?action=add last;
    }
}

**Reply by thorsten** [reply, marked as answer]
The rewrite rules for nginx can be found in this file: https://github.com/thorsten/phpMyFAQ/blob/3.2/nginx.conf 

Do you use it?

.htaccess is for Apache only.

**Reply by NoipName** [reply]
That's works thanks you !!!!! 

Can you add this section for the "defaut nginx configuration" in documentation of installation guide ? :D 

Thanks you

**Reply by thorsten** [reply]
It's documentated: https://phpmyfaq.readthedocs.io/en/3.2/installation/#nginx-web-server
